osi_logicallane.proto

```diff
@@ -0,0 +1,502 @@
+    // \c false means driving direction is according to descending S position.
+    //
+    optional bool reference_line_is_driving_direction = 8;
+
+    // Lanes that are directly right of this lane, without gap or overlap.
+    // "Right" is in definition direction (not driving direction), so right lanes
+    // have smaller T coordinates.
+    // Entries must be ordered: first by start_s, then by end_s.
+    //
+    repeated LaneRelation right_adjacent_lane = 9;
+
+    // Lanes that are directly left of this lane, without gap or overlap.
+    // "Left" is in definition direction (not driving direction), so left lanes
+    // have larger T coordinates.
+    // Entries must be ordered: first by start_s, then by end_s.
+    //
+    repeated LaneRelation left_adjacent_lane = 10;
+
+    // Lanes that partially or completely overlap this lane. Only overlaps
+    // laterally larger than 5cm are considered overlaps for the purpose of
+    // this relation.
+    // Entries must be ordered: first by start_s, then by end_s.
+    //
+    repeated LaneRelation overlapping_lane = 11;
+
+    // Right boundary of this lane.
+    // References to LogicalLaneBoundary elements.
+    // All LogicalLaneBoundary elements referenced here must refer to the same
+    // ReferenceLine as this lane.
+    // The boundaries together must cover the whole length of the lane (the
+    // range [start_s,end_s]) without gap or overlap. The boundaries must be
+    // stored in ascending order, starting with the smallest S position.
+    // Consecutive boundaries must share a point: the last point of the
+    // previous boundary must be identical to the first point of the next
+    // boundary.
+    //
+    // TODO Is it allowed for the boundaries to be longer than the lane?
+    // Theoretically, this could be done. Then the user would have to limit the
+    // boundary itself to [start_s,end_s].
+    //
+    repeated Identifier right_boundary_id = 12;
+
+    // Left boundary of this lane.
+    // References to LogicalLaneBoundary elements.
+    // All LogicalLaneBoundary elements referenced here must refer to the same
+    // ReferenceLine as this lane.
+    // The boundaries together must cover the whole length of the lane (the
+    // range [start_s,end_s]) without gap or overlap. The boundaries must be
+    // stored in ascending order, starting with the smallest S position.
+    // Consecutive boundaries must share a point: the last point of the
+    // previous boundary must be identical to the first point of the next
+    // boundary.
+    //
+    // TODO Is it allowed for the boundaries to be longer than the lane?
+    // Theoretically, this could be done. Then the user would have to limit the
+    // boundary itself to [start_s,end_s].
+    //
+    repeated Identifier left_boundary_id = 13;
+
+    // Lanes that directly are connected to this lane at the beginning
+    //
+    repeated LaneConnection predecessor_lane = 14;
```

Comment:
Just a suggestion to shorten code and reuse things: Couldn't you just take `repeated Lane.Classification.LanePairing` with a comment that logical lanes should be used? And skip everything till line 462? I am not sure if this suggestion would be better: I think it would definitely be nicer if the fields in the LanePairing message itself were repeated (and then using it with `optional`).

Reply:
`LanePairing` cannot be used, because it doesn't identify where the lanes are connected.

Example 1 (arrow describes the reference line direction):

```
 ---------------  ---------------
   -> lane 1            -> lane 2
 --------------- ----------------
```

Example 2:
```
 ---------------  ---------------
   -> lane 1            <- lane 2
 --------------- ----------------
```

In both examples, lane 2 is a successor of lane 1, since it is connected to lane 1 at the end of lane 1 (in reference line direction).

But in example 1, `at_begin_of_other_lane` is true, but in example 2, it is false.
This distinction is important for agent models, and cannot be expressed using `LanePairing` currently.

Reply:
After yesterday's discussion I have a better feeling deviating from `Lane.Classification.LanePairing` because that definition might be dropped anyhow at some point due to ambiguities as you point out. Also I actually like your approach better, because predecessor and successor lanes a separated fields which make predecessor/successor relationships easier for some roads like this one:
```
---------------\                                       /----------------------
   ->   l1       \------------ -----------------------/          ->    l2
 ----------                    ->        l3                -----------------
   ->   l4      /----------------- -------- ----------\           ->    l5
  ------------ /                                        \-----------------------
```

Reply:
@FabianPfeufferCAP for your awareness. This new suggestion might make the current LanePairing a candidate for deprecation, especially if the centerline gets dropped. Would this suggestion here also be a solution to your issues raised in https://github.com/OpenSimulationInterface/open-simulation-interface/pull/542?